kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableFeatures.java

```diff
@@ -48,19 +45,19 @@ public class TableFeatures {
   ////////////////////
 
   public static void validateReadSupportedTable(
-      Protocol protocol, Metadata metadata, String tablePath) {
+      Protocol protocol, Optional<Metadata> metadata, String tablePath) {
```

Comment:
can we just add a method that checks readerFeatures doesn't contain that Kernel doesn't know. Similar to `SUPPORTED_WRITER_FEATURES`, add `SUPPORTED_READER_FEATURES`?

Reply:
with that this method can also be refactored to use the set and then do specific metadata checks based on the readerFEatuer (e.g. column mapping)

Reply:
Factored out the `SUPPORTED_READER_FEATURES`. I think it still seems fine to me to include the `metadata` as an optional parameter; if it's present we do the metadata checks.

Separate fx would still need to check versions + the features.